src/theme-provider.js

```diff
@@ -16,6 +19,16 @@ class ThemeProvider extends Component {
   // create theme, by merging with outer theme, if present
   getTheme(passedTheme) {
     const theme = passedTheme || this.props.theme
+    if (isFunction(theme)) {
+      const mergedTheme = theme(this.outerTheme)
+      if (!isPlainObject(mergedTheme)) {
+        throw new Error(
+          '[ThemeProvider] Please return an object from your theme function, ' +
```

Comment:
Kinda like the tag idea here. Do we think it would be useful for the tag mention `glamorous` in here for further visibility into the problem? `[glamorous.ThemeProvider]`, `[glamorous] ThemProvider - ..` etc